collector/cache.go

```diff
@@ -0,0 +1,452 @@
+			nil,
+			nil,
+		),
+		FastReadNotPossiblesTotal: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "fast_read_not_possibles_total"),
+			"(FastReadNotPossiblesTotal)",
+			nil,
+			nil,
+		),
+		FastReadResourceMissesTotal: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "fast_read_resource_misses_total"),
+			"(FastReadResourceMissesTotal)",
+			nil,
+			nil,
+		),
+		FastReadsTotal: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "fast_reads_total"),
+			"(FastReadsTotal)",
+			nil,
+			nil,
+		),
+		LazyWriteFlushesTotal: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "lazy_write_flushes_total"),
+			"(LazyWriteFlushesTotal)",
+			nil,
+			nil,
+		),
+		LazyWritePagesTotal: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "lazy_write_pages_total"),
+			"(LazyWritePagesTotal)",
+			nil,
+			nil,
+		),
+		MDLReadHitsTotal: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "mdl_read_hits_total"),
+			"(MDLReadHitsTotal)",
+			nil,
+			nil,
+		),
+		MDLReadsTotal: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "mdl_reads_total"),
+			"(MDLReadsTotal)",
+			nil,
+			nil,
+		),
+		PinReadHitsTotal: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "pin_read_hits_total"),
+			"(PinReadHitsTotal)",
+			nil,
+			nil,
+		),
+		PinReadsTotal: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "pin_reads_total"),
+			"(PinReadsTotal)",
+			nil,
+			nil,
+		),
+		ReadAheadsTotal: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "read_aheads_total"),
+			"(ReadAheadsTotal)",
+			nil,
+			nil,
+		),
+		SyncCopyReadsTotal: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "sync_copy_reads_total"),
+			"(SyncCopyReadsTotal)",
+			nil,
+			nil,
+		),
+		SyncDataMapsTotal: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "sync_data_maps_total"),
+			"(SyncDataMapsTotal)",
+			nil,
+			nil,
+		),
+		SyncFastReadsTotal: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "sync_fast_reads_total"),
+			"(SyncFastReadsTotal)",
+			nil,
+			nil,
+		),
+		SyncMDLReadsTotal: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "sync_mdl_reads_total"),
+			"(SyncMDLReadsTotal)",
+			nil,
+			nil,
+		),
+		SyncPinReadsTotal: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "sync_pin_reads_total"),
+			"(SyncPinReadsTotal)",
+			nil,
+			nil,
+		),
+	}, nil
+}
+
+// Collect implements the Collector interface
+func (c *CacheCollector) Collect(ctx *ScrapeContext, ch chan<- prometheus.Metric) error {
+	if desc, err := c.collect(ctx, ch); err != nil {
+		log.Error("failed collecting cache metrics:", desc, err)
+		return err
+	}
+	return nil
+}
+
+// Win32_PerfFormattedData_PerfOS_Cache  docs:
+// - https://docs.microsoft.com/en-us/previous-versions/aa394267(v=vs.85)
```

Comment:
Do we have a better link?

Reply:
Unfortunately no. This is the only official documentation for the class I've been able to find, so @wbagdon had it right when using this link for documentation.